### PR TITLE
24 refactor authentication

### PR DESCRIPTION
**Pull request by KevinMi2023p**
Authentication has been refactored to use AWS STS, which allows the user to check if they've entered valid keys without configuring the bucket.

In the front end, a user can log in without typing anything into the bucket. This is a front end change that we're going to have to do later.

### CODE REVIEW

File: src/aws_s3_auth_conn.tsx [hunk 1]
```diff
@@ -59,11 +86,23 @@ export class MyS3Auth {
...
+      setViewState(args);
+    }).catch((err) => {
+      console.log(err)
+    })
+  }
+
+
+  // Check if the user's keys are valid
+  // Code after this function call will likely execute before this function finishes
+  checkAndDisplayValidUser(setViewState : (args : string) => any, args : string, setS3Obj : (s3Obj : MyS3Auth) => any, s3Obj : MyS3Auth ) {
```
Comment (kpattaswamy) [inline src/aws_s3_auth_conn.tsx:103]: strongly type return types

File: src/aws_s3_auth_conn.tsx [hunk 2]
```diff
@@ -59,11 +86,23 @@ export class MyS3Auth {
   // Check if the user's keys are valid for specified bucket
   // Save the S3 object made upon validation
   // Code after this function call will likely execute before this function finishes
-  checkAndDisplayValidUser(setViewState : (args : string) => any, args : string, setS3Obj : (s3Obj : MyS3Auth) => any, s3Obj : MyS3Auth ) {
+  checkAndDisplayValidBucket(setViewState : (args : string) => any, args : string, setS3Obj : (s3Obj : MyS3Auth) => any, s3Obj : MyS3Auth ) {
```
Comment (kpattaswamy) [inline src/aws_s3_auth_conn.tsx:89]: function name probably not super representative of functionality

**Review by kpattaswamy — COMMENTED** [review]
nice!

File: src/aws_s3_auth_conn.tsx (hunk 1 above)
Comment (KevinMi2023p) [reply src/aws_s3_auth_conn.tsx:103]: that's issue #44

File: src/aws_s3_auth_conn.tsx (hunk 2 above)
Comment (KevinMi2023p) [reply src/aws_s3_auth_conn.tsx:89]: Said to leave the stale function

**Review by kpattaswamy — APPROVED** [review]

**Pull request merged**